x-pack/plugins/apm/server/routes/status_check/agent_status_check.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { APMEventClient } from '../../lib/helpers/create_es_client/create_apm_event_client';
+
+export interface AgentStatusCheckResponse {
+  status: boolean;
+}
+
+export async function checkAgentStatus({
+  apmEventClient,
+}: {
+  apmEventClient: APMEventClient;
+}): Promise<AgentStatusCheckResponse> {
+  const params = {
+    apm: {
+      events: [
+        ProcessorEvent.transaction,
+        ProcessorEvent.error,
+        ProcessorEvent.metric,
+      ],
+    },
+    body: {
+      track_total_hits: true,
+      size: 1,
+      query: {
+        bool: {
+          filter: [
+            {
+              terms: {
+                'processor.event': ['error', 'transaction', 'metric'],
+              },
+            },
+          ],
+        },
+      },
```

Comment:
I'm wondering if you need this filter. Is there any data in those indices that we want to exclude?

Reply:
TBH, i simply replicated the current query we were passing to Home Tutorial App for this status check

https://github.com/achyutjhunjhunwala/kibana/blob/00e45c64a4e0121e00ea94774280c39f58044f5a/x-pack/plugins/apm/server/tutorial/envs/on_prem.ts#L128-L138
https://github.com/elastic/kibana/blob/00e45c64a4e0121e00ea94774280c39f58044f5a/x-pack/plugins/apm/server/tutorial/envs/on_prem.ts#L128-L138